src/monster.cpp

```diff
@@ -458,12 +452,124 @@ int monster::print_info(WINDOW* w, int vStart, int vLines, int column) const
+    }
+    if( get_size() == MS_SMALL ) {
+        size_name = _( "small" );
+    }
+    if( get_size() == MS_MEDIUM ) {
+        size_name = _( "medium" );
+    }
+    if( get_size() == MS_LARGE ) {
+        size_name = _( "large" );
+    }
+    if( get_size() == MS_HUGE ) {
+        size_name = _( "huge" );
+    }
+    ss << string_format( _( "It is %s in size." ), size_name.c_str() ) << std::endl;
+
+    std::vector<std::string> types;
+    if( type->has_flag( MF_ANIMAL ) ) {
+        types.emplace_back( _( "animal") );
+    }
+    if( type->in_species( ZOMBIE ) ) {
+        types.emplace_back( _( "zombie" ) );
+    }
+    if( type->in_species( FUNGUS ) ) {
+        types.emplace_back( _( "fungus" ) );
+    }
+    if( type->in_species( INSECT ) ) {
+        types.emplace_back( _( "insect" ) );
+    }
+    if( type->in_species( ABERRATION ) ) {
+        types.emplace_back( _( "aberration" ) );
+    }
+    if( !types.empty() ) {
+        ss << string_format( _( "It is %s." ),
+                             enumerate_as_string( types ).c_str() ) << std::endl;
+    }
+
+
+    std::vector<std::string> senses;
+    if( type->has_flag( m_flag::MF_HEARS ) ) {
+        senses.emplace_back( _( "hearing" ) );
+    }
+    if( type->has_flag( m_flag::MF_SEES ) ) {
+        senses.emplace_back( _( "sight" ) );
+    }
+    if( type->has_flag( m_flag::MF_SMELLS ) ) {
+        senses.emplace_back( _( "smell" ) );
+    }
+    if( senses.empty() ) {
+        ss << _( "It doesn't have senses." ) << std::endl;
+    } else {
+        ss << string_format( _( "It has the following senses: %s." ),
+                             enumerate_as_string( senses ).c_str() ) << std::endl;
+    }
+
+    std::vector<std::string> abilities;
+    if( type->has_flag( m_flag::MF_SWIMS ) ) {
+        abilities.emplace_back( _( "swim" ) );
+    }
+    if( type->has_flag( m_flag::MF_FLIES ) ) {
+        abilities.emplace_back( _( "fly" ) );
+    }
+    if( type->has_flag( m_flag::MF_CAN_DIG ) ) {
+        abilities.emplace_back( _( "dig" ) );
+    }
+    if( type->has_flag( m_flag::MF_CLIMBS ) ) {
+        abilities.emplace_back( _( "climb" ) );
+    }
+    if( !abilities.empty() ) {
+        ss << string_format( _( "It can %s." ),
+                             enumerate_as_string( abilities ).c_str() ) << std::endl;
+    }
+
+    std::vector<std::string> dangers;
+    if( type->has_flag( m_flag::MF_GRABS ) ) {
+        dangers.emplace_back( _( "grab" ) );
+    }
+    if( type->has_flag( m_flag::MF_VENOM ) ) {
+        dangers.emplace_back( _( "poison" ) );
+    }
+    if( type->has_flag( m_flag::MF_PARALYZE ) ) {
+        dangers.emplace_back( _( "paralyze" ) );
+    }
+    if( type->has_flag( m_flag::MF_BLEED ) ) {
+        dangers.emplace_back( _( "cause bleed" ) );
+    }
+    if( !dangers.empty() ) {
+        ss << string_format( _( "<bad>In fight it can %s.</bad>" ),
+                             enumerate_as_string( dangers ).c_str() ) << std::endl;
+    }
```

Comment:
You're repeating this kind of block 3 times.
Would be much better if you cleaned it up so that there would be copies of clean code, not "raw" `if` blocks.

For example:
```
using flag_description = std::pair<mflag, std::string>;
...
const auto describe_flags = [this]( stringstream &ss, const std::string &format, const std::vector<flag_description> &flags_names ) {
    std::string flag_descriptions = enumerate_as_string( flags_names.begin(), flags_names.end(), [this]( const flag_description &fd ) {
        return type->has_flag( fd.first ) ? fd.second : "";
    } );
    if( !flag_descriptions.empty() ) {
        ss << string_format( format, flag_descriptions.c_str() ) << std::endl;
    }
};
...
std::vector<flag_description> danger_flags {{
    { mflag::MF_VENOM, translate_marker( "poison" ) },
    { mflag::MF_BLEED, translate_marker( "cause bleed" ) },
    ...
}};
describe_flags( ss, _( "<bad>In fight it can %s.</bad>" ), danger_flags );
...
std::vector<flag_description> movement_flags {{
   { mflag::DIGS, ...
}};
describe_flags( "it can %s", movement_flags );
```

Wrapping everything in functions helps readability: entire page has only dense data-to-text relation, no padding or boilerplate. If the functions are named correctly, it is clear from context what will happen.